diff --git a/setup.py b/setup.py
index 91c24d91..58c7da57 100755
--- a/setup.py
+++ b/setup.py
@@ -14,9 +14,6 @@
 here = path.abspath(path.dirname(__file__))


-os.environ["CC"] = "g++"
-os.environ["CXX"] = "g++"
-
 # dirty but working
 __version__ = re.search(
     r'__version__\s*=\s*[\'"]([^\'"]*)[\'"]',  # It excludes inline comment too
@@ -39,7 +36,9 @@


 # add platform dependant optional compilation argument
-if sys.platform.startswith('darwin'):
+if sys.platform.startswith('darwin') and not os.environ.get('POT_LEAVE_CC', ''):
+    os.environ["CC"] = "g++"
+    os.environ["CXX"] = "g++"
     opt_arg.append("-stdlib=libc++")
     sdk_path = subprocess.check_output(['xcrun', '--show-sdk-path'])
     os.environ['CFLAGS'] = '-isysroot "{}"'.format(sdk_path.rstrip
